libs/dal/src/repositories/topic/topic.repository.ts

```diff
@@ -30,14 +31,41 @@ export class TopicRepository extends BaseRepository<EnforceEnvironmentQuery, Top
   }
 
   async findTopic(entity: Omit<TopicEntity, 'key' | 'name'>): Promise<TopicEntity> {
+    const TOPIC_SUBSCRIBERS_COLLECTION = 'topicsubscribers';
+
     const { _environmentId, _id, _organizationId, _userId } = entity;
 
-    return await this.findOne({
-      _environmentId,
-      _id,
-      _organizationId,
-      _userId,
-    });
+    const [result] = await this.aggregate([
+      {
+        $match: { _organizationId, _environmentId, _id, _userId },
+      },
+      {
+        $lookup: {
+          from: TOPIC_SUBSCRIBERS_COLLECTION,
+          localField: '_id',
+          foreignField: '_topicId',
+          as: 'topicSubscribers',
+        },
+      },
+      {
+        $project: {
+          _environmentId: 1,
+          _organizationId: 1,
+          _userId: 1,
+          key: 1,
+          name: 1,
+          // The lookup returns a matrix so we return the first array element in the projection
+          subscribers: { $arrayElemAt: ['$topicSubscribers.subscribers', 0] },
+        },
+      },
+      { $limit: 1 },
+    ]);
```

Comment:
A small question, why are we doing this instead of just doing a regular `find` method? Just to return the list of subscriber on the DTO? I would actually suggest not to return it in the find topic, and have a separate method to find topic subscribers that will be standalone, because the topic subscribers potential can be thousands of people and might require pagination and etc...